sql/executor.go

```diff
@@ -678,8 +679,10 @@ func checkResultDatum(datum parser.Datum) error {
 	case parser.DDate:
 	case parser.DTimestamp:
 	case parser.DInterval:
+	case parser.DValArg:
+		return fmt.Errorf("could not determine data type of parameter %s", d)
```

Comment:
Ah, nice. If you wanted to golf a bit more, you could even get rid of `d` and just use `datum` here.